### PR TITLE
[IGNORE] fix: add new Bar component

### LINKED ISSUE #1956: Bar component adoption
There is a new component in Fundamental-styles - Bar. It's used as a standalone component or as a component to build headers, subheaders, footers in Popovers and Modals.

### PR DESCRIPTION
**Pull request by InnaAtanasova**
#### Please provide a link to the associated issue.
closes #1956 
#### Please provide a brief summary of this pull request.
NGX implementation of Bar component released in Fundamental-styles 0.6.0
#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done

Documentation checklist:
- [x] update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

**Comment by mikerodonnell89** [comment]
Is there supposed to be a difference between the first two examples (default bar and header bar)? I see the code is different but the examples look the same

**Comment by mikerodonnell89** [comment]
Why would mobile/tablet mode be an input and not based off screen size?

File: libs/core/src/lib/bar/directives/bar-element.directive.spec.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    template: `
+        <div #directiveElement fd-bar-element [fullWidth]="true">Bar Element Test</div>
+    `
+})
+class TestComponent {
+    @ViewChild('directiveElement', { static: false })
+    ref: ElementRef;
+}
+
+describe('ActionBarActionsDirective', () => {
```
Comment (mikerodonnell89) [inline libs/core/src/lib/bar/directives/bar-element.directive.spec.ts]: Incorrect name, applies to other tests in this pr as well

File: libs/core/src/lib/bar/bar.component.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import {
+    AfterViewInit, ChangeDetectionStrategy, Component, ElementRef, HostBinding, Input, ViewEncapsulation
+} from '@angular/core';
+import { CssClassBuilder, applyCssClass } from '../utils/public_api';
```
Comment (JKMarkowski) [inline libs/core/src/lib/bar/bar.component.ts]: Let's remove unused imports

File: libs/core/src/lib/bar/directives/bar-right.directive.spec.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    template: `
+        <div #directiveElement fd-bar-right>Bar Right Area Test</div>
+    `
+})
+class TestComponent {
+    @ViewChild('directiveElement', { static: false })
+    ref: ElementRef;
+}
+
+describe('ActionBarActionsDirective', () => {
```
Comment (JKMarkowski) [inline libs/core/src/lib/bar/directives/bar-right.directive.spec.ts]: Let's take a look again on test names, there are a lot of wrong named

**Comment by InnaAtanasova** [comment]
> Is there supposed to be a difference between the first two examples (default bar and header bar)? I see the code is different but the examples look the same

I agree, it's just in the documentation (wiki specs) there's `auto` and `header` types in BarDesign. I followed the specs in case the design changes in future and introduce different values for the header.

File: libs/core/src/lib/bar/directives/bar-element.directive.spec.ts (hunk 1 above)
Comment (InnaAtanasova) [reply libs/core/src/lib/bar/directives/bar-element.directive.spec.ts]: Thanks for noticing this. I fixed it.

**Comment by InnaAtanasova** [comment]
> Why would mobile/tablet mode be an input and not based off screen size?

To allow the user to specify their own breaking points

File: libs/core/src/lib/bar/bar.component.ts (hunk 2 above)
Comment (InnaAtanasova) [reply libs/core/src/lib/bar/bar.component.ts]: Removed

File: libs/core/src/lib/bar/directives/bar-right.directive.spec.ts (hunk 3 above)
Comment (InnaAtanasova) [reply libs/core/src/lib/bar/directives/bar-right.directive.spec.ts]: Done

**Comment by InnaAtanasova** [comment]
> LGTM, but we could improve RTL examples, or change button icon to some universal one.
> 
> image

You are right, I added styling to deal with icon RTL. Thanks for noticing it.

**Comment by InnaAtanasova** [comment]
> linter throws error
> 
> ```
> ERROR: fundamental-ngx/libs/core/src/lib/bar/directives/bar-element.directive.ts:7:15 - The selector should be used as an attribute (https://angular.io/guide/styleguide#style-02-06)
> ERROR: fundamental-ngx/libs/core/src/lib/bar/directives/bar-left.directive.ts:7:15 - The selector should be camelCased (https://angular.io/guide/styleguide#style-02-06)
> ERROR: fundamental-ngx/libs/core/src/lib/bar/directives/bar-middle.directive.ts:7:15 - The selector should be camelCased (https://angular.io/guide/styleguide#style-02-06)
> ERROR: fundamental-ngx/libs/core/src/lib/bar/directives/bar-right.directive.ts:7:15 - The selector should be camelCased (https://angular.io/guide/styleguide#style-02-06)
> ```
I made Linter happy again. Please take a look at the PR again. Thanks

File: libs/core/src/lib/bar/bar.component.html [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<div class="fd-bar"
+    [ngClass]="
+        (cosy ? ' fd-bar--cosy' : '') + 
+        (barDesign ? ' fd-bar--' + barDesign : '') + 
+        (inPage && !size ? ' fd-bar--page': '') + 
+        (inPage && size ? ' fd-bar--page-' + size: '') + 
+        (inHomePage && !size ? ' fd-bar--home-page': '') + 
+        (inHomePage && size ? ' fd-bar--home-page-' + size: '')">
```
Comment (JKMarkowski) [inline libs/core/src/lib/bar/bar.component.html]: Could you consider using `applyCssClass` decorator for that case? There is a lot of statements.

File: libs/core/src/lib/bar/bar.component.html [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<div class="fd-bar"
```
Comment (JKMarkowski) [inline libs/core/src/lib/bar/bar.component.html]:
```suggestion
<div #barElement >
```

File: libs/core/src/lib/bar/bar.component.html (hunk 4 above)
Comment (JKMarkowski) [reply libs/core/src/lib/bar/bar.component.html]:
```typescript
    @applyCssClass
    /** CssClassBuilder interface implementation
     * function must return single string
     * function is responsible for order which css classes are applied
     */
    buildComponentCssClass(): string {
        return [
            'fd-bar',
            this.cosy ? 'fd-bar--cosy' : '',
           this.barDeisgn ? ('fd-bar--' + this.badDesign) : ''
            ....
        ].filter(x => x !== '').join(' ');
    }
```

File: libs/core/src/lib/bar/bar.component.html (hunk 5 above)
Comment (JKMarkowski) [reply libs/core/src/lib/bar/bar.component.html]: @ViewChild('barElement', {}) barElement: ElementRef

File: libs/core/src/lib/bar/bar.component.html (hunk 4 above)
Comment (JKMarkowski) [reply libs/core/src/lib/bar/bar.component.html]: elementRef(): ElementRef<any> {
        return this.barElement;
    }

**Comment by InnaAtanasova** [comment]
Keeping this PR just for reference (for the comments). Will close it when merge the other one. 
Please refer to [THIS PR](https://github.com/SAP/fundamental-ngx/pull/2077)

**Pull request closed without merging**